core/arch/arm/kernel/boot.c

```diff
@@ -113,6 +113,14 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
+/* May be overridden in plat-$(PLATFORM)/main.c */
+__weak unsigned long plat_get_random_stack_canary(void)
```

Comment:
I wonder if this function shouldn't be mandatory for platforms to implement if configured with `CFG_CORE_STACK_PROTECTOR`. Without a random value we could just as well use the default value in `lib/libutils/isoc/stack_check.c`.

Reply:
Since we move the set random canary call to after boot_init_primary_late() we can use the crypto_rng_read() to generate stack canary value and doesn't need platform specific implementation. Fixed in updated patch

Reply:
This change currently broke the virtualization due to boot flow differences. `init_tee_runtime()` doesn't get call in `boot_init_primary_late()` when virtualization is enable.